### ISSUE TITLE
Version managing pandoc

### ISSUE THREAD

**Issue by ickc** [sergiocorreia/panflute#153]
c.f. https://github.com/sergiocorreia/panflute/issues/142#issuecomment-723522831

Regarding the lookup table between panflute and pandoc versions, for now it is just https://github.com/sergiocorreia/panflute/tree/stable#supported-pandoc-versions in the stable branch. It is an information for the end users and has to be acted upon manually.

Long term I'd like to have proper version control using conda. c.f. https://github.com/conda-forge/panflute-feedstock currently maintained by @kiwi0fruit. The pandoc dependence should be added there. And if done correctly people can do `conda install panflute` to have the matching version of compatible pandoc, panflute. (And say `conda install pantable` would have matching version of compatible pandoc, panflute, pantable and so on.)

For pip users, another option would be for panflute to bundle pandoc inside its wheel. I know that pypandoc used to do it in the past but has too much trouble that we wrote a download_pandoc function to just inform the endusers to run that function. However, it seems that they are building it from the source. Bundling the pre-compiled binaries in the wheels probably is much easier.

Personally I'll just focused on conda since this is what I used exclusively.

More on conda:

It is a favorite package manager in Data Science. e.g. Jupyter Notebooks and RMarkdown both uses pandoc behind the scene for conversions and can be version managed via conda. Conda also support many platforms including all those supported by pandoc. Lastly, pandoc, pandocfilters, panflute is already available there.

Edit: TODO:

- [x] constrain minimum pandoc version in conda-forge: https://github.com/conda-forge/panflute-feedstock/pull/8
- [x] constrain maximum pandoc version in conda-forge: https://github.com/conda-forge/panflute-feedstock/pull/9
- [x] document this: in which order?
    - `conda install -c conda-forge pandoc panflute` <- prefer this as this is "unbreakable"
    - `pip install panflute`

**Comment by fralau** [comment]
Thanks for opening the issue !

**Comment by ickc** [comment]
c.f. https://github.com/conda-forge/panflute-feedstock/pull/8

@kiwi0fruit, @beckermr, could you help merging it please?

**Comment by ickc** [comment]
Thanks @kiwi0fruit, @beckermr, it has been merged.

Now people can just `conda install -c conda-forge panflute pandoc` and expect the 2 versions to be matching. But when I started that PR I wasn't careful enough to require pandoc<= a certain version. So there's a new PR in https://github.com/conda-forge/panflute-feedstock/pull/9

Next step would be document it. If there's no objection I'll recommend `conda install -c conda-forge pandoc panflute` since it is unbreakable (actually not true... since they can override that in PATH...) first and mention `pip install panflute` 2nd.

Another idea is to detect a copy of pandoc along-side panflute and prefer that first (in convert_text which calls run_pandoc.) But panflute as a filter still doesn't have control on which pandoc they use (`pandoc -F panflute ...`) so may be just honor their PATH.
